lib/broccoli/ember-app.js

```diff
@@ -1180,29 +1180,35 @@ class EmberApp {
     @return {Array}
    */
   lintTestTrees() {
-    let lintedApp = this.addonLintTree('app', this._filterAppTree());
-    let lintedTests = this.addonLintTree('tests', this.trees.tests);
-    let lintedTemplates = this.addonLintTree('templates', this._templatesTree());
+    let testSupportTree = this._mergeTrees(this.addonTreesFor('test-support'));
 
-    lintedApp = processModulesOnly(new Funnel(lintedApp, {
-      srcDir: '/',
-      destDir: `${this.name}/tests/`,
-      annotation: 'Funnel (lint app)',
-    }), 'Babel: lintTree(app)');
+    return [
+      this.lintTestTree('app', this._filterAppTree()),
+      this.lintTestTree('tests', this.trees.tests),
+      this.lintTestTree('templates', this._templatesTree()),
+      this.lintTestTree('test-support', testSupportTree),
```

Comment:
See https://github.com/ember-cli/ember-cli/pull/6668 for more detail on the approach that I'm suggesting.